templates/newsroom/newsroom.js

```diff
@@ -0,0 +1,393 @@
+import {
+  div, section, input, span, a,
+} from '../../scripts/dom-helpers.js';
+
+// Function to fetch placeholders from JSON endpoint
+async function fetchPlaceholders() {
+  try {
+    const response = await fetch('https://main--fondationsaudemarspiguet--aemdemos.aem.page/placeholders.json');
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const data = await response.json();
+    return data; // Return the data object or empty object if not found
+  } catch (error) {
+    return {}; // Return empty object on error
+  }
+}
+
+// Function to detect current language
+function getCurrentLanguage() {
+  // Check URL path
+  const currentPath = window.location.pathname;
+  if (currentPath.startsWith('/fr/') || currentPath.includes('/fr/')) {
+    return 'fr';
+  }
+  // Default to English
+  return 'en';
+}
+
+// Function to get placeholder text with fallback
+function getPlaceholderText(placeholders, key, fallback) {
```

Comment:
again, function for placeholders already exists in `/scripts/aem.js`, should be used directly.
See it's usage in `news-landing` page: https://github.com/aemdemos/fondationsaudemarspiguet/blob/main/templates/news-landing/news-landing.js#L25